### PR TITLE
feat: internal endpoint to delete all records related to a tenant

### PR DESCRIPTION
**Pull request by merlin-northern**
Changelog: Title
Ticket: MEN-7319

### CODE REVIEW

**Comment by merlin-northern** [comment]
@mender-test-bot start pipeline --fast please

File: api/http/internal.go [hunk 1]
```diff
@@ -55,6 +55,26 @@ func (dev *internalDevice) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+// DELETE /tenants/:tenant_id
+// code: 204 - all tenant data removed
+//
+//	500 - internal server error on removal
```
Comment (kjaskiewiczz) [inline api/http/internal.go:61]: interesting formatting ;)

File: docs/internal_api.yml [hunk 2]
```diff
@@ -86,6 +86,25 @@ paths:
...
+  /tenants/{tenantId}:
+    delete:
+      tags:
+        - Internal API
+      operationId: Delete tenant
+      summary: Delete all data belonging to a given tenant.
+      parameters:
+        - in: path
+          name: tenantId
+          schema:
```
Comment (kjaskiewiczz) [inline docs/internal_api.yml:98]: do we need "schema" here?

File: api/http/internal.go (hunk 1 above)
Comment (merlin-northern) [reply api/http/internal.go:61]: I kept it consistent with the rest of the file. do you want me to reformat all?

File: docs/internal_api.yml (hunk 2 above)
Comment (merlin-northern) [reply docs/internal_api.yml:98]: this is how this whole file is made, do you want me to change everywhere?

**Pull request merged**